File: mobile/ios/Sources/StaffScreens.swift
import SwiftUI
import CoreLocation

let kDisciplines = ["CIVIL", "ELECTRICAL", "PLUMBING", "PEST_OTHER"]
let kPropertyTypes = ["APARTMENT", "HOUSE"]

func propertyTypeLabel(_ t: String) -> String {
    ["APARTMENT": "Apartment", "HOUSE": "House"][t] ?? t
}

private func dayKey(_ date: Date) -> String {
    let f = DateFormatter(); f.dateFormat = "yyyy-MM-dd"; f.timeZone = TimeZone(identifier: "UTC")
    return f.string(from: date)
}

// MARK: - New inspection (Page 1: customer & property → draft)

/// Two-page flow mirroring the web: enter the customer/property here, then move
/// to the assign-team page. Presented as a sheet; calls `onDone` after assigning.
struct NewInspectionFlow: View {
    let onDone: () -> Void
    @EnvironmentObject var auth: AuthStore
    @Environment(\.dismiss) private var dismiss

    @State private var name = ""
    @State private var phone = ""
    @State private var email = ""
    @State private var address = ""
    @State private var propertyType = "APARTMENT"
    @State private var inspectionType = "pre-purchase"
    @State private var coords: CLLocationCoordinate2D?
    @State private var showMap = false
    @State private var draftId: String?
    @State private var busy = false
    @State private var error: String?

    private var valid: Bool {
        !name.trimmingCharacters(in: .whitespaces).isEmpty &&
        !address.trimmingCharacters(in: .whitespaces).isEmpty &&
        !inspectionType.trimmingCharacters(in: .whitespaces).isEmpty
    }

    var body: some View {
        NavigationStack {
            Form {
                Section("Customer") {
                    TextField("Name", text: $name)
                    TextField("Phone", text: $phone).keyboardType(.phonePad)
                    TextField("Email", text: $email)
                        .textInputAutocapitalization(.never).keyboardType(.emailAddress)
                }
                Section("Property") {
                    TextField("Address", text: $address)
                    Button {
                        showMap = true
                    } label: {
                        Label(coords == nil ? "Pick on map" : "Change location", systemImage: "mappin.and.ellipse")
                    }
                    if let coords {
                        Text(String(format: "📍 %.5f, %.5f", coords.latitude, coords.longitude))
                            .font(.caption).foregroundStyle(.secondary)
                    }
                    Picker("Type", selection: $propertyType) {
                        ForEach(kPropertyTypes, id: \.self) { Text(propertyTypeLabel($0)).tag($0) }
                    }
                    TextField("Inspection type", text: $inspectionType)
                }
                if let error { Text(error).foregroundStyle(.red).font(.footnote) }
            }
            .navigationTitle("New Inspection")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .cancellationAction) { Button("Cancel") { dismiss() } }
                ToolbarItem(placement: .confirmationAction) {
                    Button(busy ? "…" : "Next") { Task { await createDraft() } }
                        .disabled(!valid || busy)
                }
            }
            .navigationDestination(item: $draftId) { id in
                AssignTeamView(inspectionId: id, customerName: name) {
                    dismiss(); onDone()
                }
            }
            .sheet(isPresented: $showMap) {
                MapPickerView(initial: coords) { coord, addr in
                    coords = coord
                    if let addr, address.trimmingCharacters(in: .whitespaces).isEmpty { address = addr }
                }
            }
        }
    }

    private func createDraft() async {
        busy = true; error = nil
        do {
            draftId = try await auth.createDraft(
                customerName: name, phone: phone, email: email,
                address: address, propertyType: propertyType,
                inspectionType: inspectionType,
                latitude: coords?.latitude, longitude: coords?.longitude)
        } catch {
            self.error = "Couldn't create the draft. \(error.localizedDescription)"
        }
        busy = false
    }
}

// MARK: - Assign team (Page 2: date + inspectors per discipline)

/// Reused for both the new-inspection flow and re-assigning an existing draft.
struct AssignTeamView: View {
    let inspectionId: String
    var customerName: String? = nil
    let onDone: () -> Void

    @EnvironmentObject var auth: AuthStore
    @State private var inspectors: [UserRef] = []
    @State private var date = Date()
    @State private var picked: [String: String] = [:]   // discipline -> inspectorId
    @State private var busy = false
    @State private var error: String?

    var body: some View {
        Form {
            if let customerName {
                Section { Text(customerName).font(.headline) }
            }
            Section("Schedule") {
                DatePicker("Date", selection: $date, displayedComponents: .date)
            }
            Section("Assign inspectors") {
                ForEach(kDisciplines, id: \.self) { d in
                    Picker(disciplineLabel(d), selection: Binding(
                        get: { picked[d] ?? "" },
                        set: { picked[d] = $0 })) {
                        Text("— none —").tag("")
                        ForEach(inspectors.filter { $0.discipline == d }) { i in
                            Text(i.name).tag(i.id)
                        }
                    }
                }
                Text("Pick an inspector for each discipline you need. Leave others as none.")
                    .font(.caption).foregroundStyle(.secondary)
            }
            if let error { Text(error).foregroundStyle(.red).font(.footnote) }
        }
        .navigationTitle("Assign Team")
        .navigationBarTitleDisplayMode(.inline)
        .toolbar {
            ToolbarItem(placement: .confirmationAction) {
                Button(busy ? "Saving…" : "Save") { Task { await save() } }
                    .disabled(busy)
            }
        }
        .task {
            inspectors = ((try? await auth.users()) ?? []).filter { $0.role == "INSPECTOR" }
        }
    }

    private func save() async {
        busy = true; error = nil
        let assignments = picked
            .filter { !$0.value.isEmpty }
            .map { (discipline: $0.key, inspectorId: $0.value) }
        do {
            try await auth.assignTeam(inspectionId, scheduledAt: dayKey(date), assignments: assignments)
            onDone()
        } catch {
            self.error = "Couldn't save the assignment. \(error.localizedDescription)"
        }
        busy = false
    }
}

// MARK: - Users / team management

struct UsersView: View {
    @EnvironmentObject var auth: AuthStore
    @State private var users: [UserRef] = []
    @State private var loading = true
    @State private var showAdd = false

    private var canAdd: Bool { auth.user?.role == "ADMIN" }

    var body: some View {
        List {
            if loading {
                ProgressView()
            } else {
                ForEach(users) { u in
                    VStack(alignment: .leading, spacing: 3) {
                        HStack {
                            Text(u.name).font(.headline)
                            Spacer()
                            StatusPill(text: u.role.capitalized,
                                       tone: u.role == "ADMIN" ? .issue : u.role == "MANAGER" ? .brand : .neutral)
                        }
                        Text(u.email).font(.subheadline).foregroundStyle(.secondary)
                        if let d = u.discipline { Text(disciplineLabel(d)).font(.caption).foregroundStyle(.secondary) }
                    }.padding(.vertical, 2)
                }
            }
        }
        .navigationTitle("Team")
        .toolbar {
            if canAdd {
                ToolbarItem(placement: .topBarTrailing) {
                    Button { showAdd = true } label: { Image(systemName: "person.badge.plus") }
                }
            }
        }
        .sheet(isPresented: $showAdd) { AddUserView { Task { await load() } } }
        .task { await load() }
    }

    private func load() async {
        loading = true
        users = (try? await auth.users()) ?? []
        loading = false
    }
}

struct AddUserView: View {
    let onDone: () -> Void
    @EnvironmentObject var auth: AuthStore
    @Environment(\.dismiss) private var dismiss

    @State private var name = ""
    @State private var email = ""
    @State private var password = ""
    @State private var role = "INSPECTOR"
    @State private var discipline = "CIVIL"
    @State private var busy = false
    @State private var error: String?

    private var valid: Bool {
        !name.isEmpty && email.contains("@") && password.count >= 6
    }

    var body: some View {
        NavigationStack {
            Form {
                Section("Account") {
                    TextField("Name", text: $name)
                    TextField("Email", text: $email)
                        .textInputAutocapitalization(.never).keyboardType(.emailAddress)
                    SecureField("Password (min 6)", text: $password)
                }
                Section("Role") {
                    Picker("Role", selection: $role) {
                        Text("Inspector").tag("INSPECTOR")
                        Text("Manager").tag("MANAGER")
                        Text("Admin").tag("ADMIN")
                    }
                    if role == "INSPECTOR" {
                        Picker("Discipline", selection: $discipline) {
                            ForEach(kDisciplines, id: \.self) { Text(disciplineLabel($0)).tag($0) }
                        }
                    }
                }
                if let error { Text(error).foregroundStyle(.red).font(.footnote) }
            }
            .navigationTitle("Add User")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .cancellationAction) { Button("Cancel") { dismiss() } }
                ToolbarItem(placement: .confirmationAction) {
                    Button(busy ? "…" : "Create") { Task { await create() } }
                        .disabled(!valid || busy)
                }
            }
        }
    }

    private func create() async {
        busy = true; error = nil
        do {
            try await auth.createUser(name: name, email: email, password: password,
                                      role: role, discipline: role == "INSPECTOR" ? discipline : nil)
            dismiss(); onDone()
        } catch {
            self.error = "Couldn't create the user. \(error.localizedDescription)"
        }
        busy = false
    }
}

File: mobile/android/app/src/main/java/test/check/inspections/ui/StaffScreens.kt
package test.check.inspections.ui

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material.icons.filled.LocationOn
import androidx.compose.material.icons.filled.PersonAdd
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.viewinterop.AndroidView
import kotlinx.coroutines.launch
import org.osmdroid.config.Configuration
import org.osmdroid.events.MapEventsReceiver
import org.osmdroid.tileprovider.tilesource.TileSourceFactory
import org.osmdroid.util.GeoPoint
import org.osmdroid.views.MapView
import org.osmdroid.views.overlay.MapEventsOverlay
import org.osmdroid.views.overlay.Marker
import test.check.inspections.data.*
import java.time.LocalDate

// MARK: New inspection — page 1 (customer + property + map)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun NewInspectionScreen(backend: Backend, onAssign: (String) -> Unit, onBack: () -> Unit) {
    var name by remember { mutableStateOf("") }
    var phone by remember { mutableStateOf("") }
    var email by remember { mutableStateOf("") }
    var address by remember { mutableStateOf("") }
    var propertyType by remember { mutableStateOf("APARTMENT") }
    var inspectionType by remember { mutableStateOf("pre-purchase") }
    var coords by remember { mutableStateOf<Pair<Double, Double>?>(null) }
    var showMap by remember { mutableStateOf(false) }
    var busy by remember { mutableStateOf(false) }
    var error by remember { mutableStateOf<String?>(null) }
    val scope = rememberCoroutineScope()

    if (showMap) {
        MapPicker(
            initial = coords ?: (33.3152 to 44.3661),
            onPick = { la, ln -> coords = la to ln; showMap = false },
            onClose = { showMap = false },
        )
        return
    }

    val valid = name.isNotBlank() && address.isNotBlank() && inspectionType.isNotBlank()

    Scaffold(topBar = {
        TopAppBar(
            title = { Text(tr("New Inspection")) },
            navigationIcon = { IconButton(onClick = onBack) { Icon(Icons.Default.ArrowBack, null) } },
            actions = {
                TextButton(enabled = valid && !busy, onClick = {
                    busy = true; error = null
                    scope.launch {
                        try {
                            val r = backend.api.createDraft(
                                CreateDraftBody(
                                    customer = CustomerInput(name.trim(), phone.ifBlank { null }, email.ifBlank { null }),
                                    property = PropertyInput(address.trim(), propertyType, coords?.first, coords?.second),
                                    type = inspectionType.trim(),
                                )
                            )
                            onAssign(r.id)
                        } catch (e: Exception) { error = "Couldn't create the draft." }
                        busy = false
                    }
                }) { Text(if (busy) "…" else tr("Next")) }
            }
        )
    }) { pad ->
        Column(
            Modifier.padding(pad).fillMaxSize().verticalScroll(rememberScrollState()).padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(10.dp)
        ) {
            Text(tr("Customer"), fontWeight = FontWeight.Bold)
            OutlinedTextField(name, { name = it }, label = { Text(tr("Name")) }, singleLine = true, modifier = Modifier.fillMaxWidth())
            OutlinedTextField(phone, { phone = it }, label = { Text(tr("Phone")) }, singleLine = true, modifier = Modifier.fillMaxWidth())
            OutlinedTextField(email, { email = it }, label = { Text(tr("Email")) }, singleLine = true, modifier = Modifier.fillMaxWidth())

            Spacer(Modifier.height(4.dp))
            Text(tr("Property"), fontWeight = FontWeight.Bold)
            OutlinedTextField(address, { address = it }, label = { Text(tr("Address")) }, modifier = Modifier.fillMaxWidth())
            OutlinedButton(onClick = { showMap = true }, modifier = Modifier.fillMaxWidth()) {
                Icon(Icons.Default.LocationOn, null); Spacer(Modifier.width(6.dp))
                Text(if (coords == null) tr("Pick on map") else tr("Change location"))
            }
            coords?.let { Text(tr("📍 %.5f, %.5f").format(it.first, it.second), color = Color.Gray, style = MaterialTheme.typography.bodySmall) }

            EnumDropdown("Type", PROPERTY_TYPES, propertyType, ::propertyTypeLabel) { propertyType = it }
            OutlinedTextField(inspectionType, { inspectionType = it }, label = { Text(tr("Inspection type")) }, singleLine = true, modifier = Modifier.fillMaxWidth())
            error?.let { Text(it, color = IssueRed, style = MaterialTheme.typography.bodySmall) }
        }
    }
}

// MARK: Map picker (OpenStreetMap via osmdroid — no API key)

@Composable
fun MapPicker(initial: Pair<Double, Double>, onPick: (Double, Double) -> Unit, onClose: () -> Unit) {
    val context = LocalContext.current
    var lat by remember { mutableStateOf(initial.first) }
    var lng by remember { mutableStateOf(initial.second) }

    LaunchedEffect(Unit) {
        Configuration.getInstance().userAgentValue = context.packageName
    }

    Column(Modifier.fillMaxSize()) {
        Box(Modifier.weight(1f)) {
            AndroidView(factory = { ctx ->
                MapView(ctx).apply {
                    setTileSource(TileSourceFactory.MAPNIK)
                    setMultiTouchControls(true)
                    controller.setZoom(13.0)
                    controller.setCenter(GeoPoint(lat, lng))
                    val marker = Marker(this).apply { position = GeoPoint(lat, lng) }
                    overlays.add(marker)
                    val receiver = object : MapEventsReceiver {
                        override fun singleTapConfirmedHelper(p: GeoPoint?): Boolean {
                            p?.let { lat = it.latitude; lng = it.longitude; marker.position = it; invalidate() }
                            return true
                        }
                        override fun longPressHelper(p: GeoPoint?) = false
                    }
                    overlays.add(0, MapEventsOverlay(receiver))
                }
            }, modifier = Modifier.fillMaxSize())
        }
        Surface(tonalElevation = 3.dp) {
            Column(Modifier.fillMaxWidth().padding(14.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
                Text(tr("Tap the map to drop a pin"), color = Color.Gray, style = MaterialTheme.typography.bodySmall)
                Text(tr("📍 %.5f, %.5f").format(lat, lng), fontWeight = FontWeight.SemiBold)
                Row(verticalAlignment = Alignment.CenterVertically) {
                    TextButton(onClick = onClose) { Text(tr("Cancel")) }
                    Spacer(Modifier.weight(1f))
                    Button(onClick = { onPick(lat, lng) }) { Text(tr("Use this location")) }
                }
            }
        }
    }
}

// MARK: Assign team — page 2 (date + inspectors per discipline)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AssignTeamScreen(backend: Backend, id: String, onDone: () -> Unit, onBack: () -> Unit) {
    var inspectors by remember { mutableStateOf<List<UserRef>>(emptyList()) }
    var date by remember { mutableStateOf(LocalDate.now().toString()) }
    val picked = remember { mutableStateMapOf<String, String>() } // discipline -> inspectorId
    var busy by remember { mutableStateOf(false) }
    var error by remember { mutableStateOf<String?>(null) }
    val scope = rememberCoroutineScope()

    LaunchedEffect(Unit) {
        inspectors = try { backend.api.users().filter { it.role == "INSPECTOR" } } catch (e: Exception) { emptyList() }
    }

    Scaffold(topBar = {
        TopAppBar(
            title = { Text(tr("Assign Team")) },
            navigationIcon = { IconButton(onClick = onBack) { Icon(Icons.Default.ArrowBack, null) } },
            actions = {
                TextButton(enabled = !busy, onClick = {
                    busy = true; error = null
                    scope.launch {
                        try {
                            val assignments = picked.filterValues { it.isNotBlank() }
                                .map { (d, i) -> AssignInput(d, i) }
                            backend.api.assignTeam(id, AssignTeamBody(scheduledAt = date, assignments = assignments))
                            onDone()
                        } catch (e: Exception) { error = "Couldn't save the assignment." }
                        busy = false
                    }
                }) { Text(if (busy) tr("Saving…") else tr("Save")) }
            }
        )
    }) { pad ->
        Column(
            Modifier.padding(pad).fillMaxSize().verticalScroll(rememberScrollState()).padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Text(tr("Schedule"), fontWeight = FontWeight.Bold)
            OutlinedTextField(date, { date = it }, label = { Text(tr("Date (YYYY-MM-DD)")) }, singleLine = true, modifier = Modifier.fillMaxWidth())

            Text(tr("Assign inspectors"), fontWeight = FontWeight.Bold)
            DISCIPLINES.forEach { d ->
                val options = inspectors.filter { it.discipline == d }
                InspectorDropdown(disciplineLabel(d), options, picked[d]) { picked[d] = it ?: "" }
            }
            Text(tr("Pick an inspector for each discipline you need. Leave others empty."),
                color = Color.Gray, style = MaterialTheme.typography.bodySmall)
            error?.let { Text(it, color = IssueRed, style = MaterialTheme.typography.bodySmall) }
        }
    }
}

// MARK: Users / team

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun UsersScreen(backend: Backend, onBack: () -> Unit) {
    val user by backend.user.collectAsState()
    val canAdd = user?.role == "ADMIN"
    var users by remember { mutableStateOf<List<UserRef>>(emptyList()) }
    var loading by remember { mutableStateOf(true) }
    var showAdd by remember { mutableStateOf(false) }
    val scope = rememberCoroutineScope()

    suspend fun load() {
        loading = true
        users = try { backend.api.users() } catch (e: Exception) { emptyList() }
        loading = false
    }
    LaunchedEffect(Unit) { load() }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text(tr("Team")) },
                navigationIcon = { IconButton(onClick = onBack) { Icon(Icons.Default.ArrowBack, null) } },
                actions = { if (canAdd) IconButton(onClick = { showAdd = true }) { Icon(Icons.Default.PersonAdd, "Add user") } }
            )
        }
    ) { pad ->
        if (loading) {
            Box(Modifier.fillMaxSize().padding(pad), Alignment.Center) { CircularProgressIndicator() }
        } else {
            LazyColumn(Modifier.padding(pad).fillMaxSize(), contentPadding = PaddingValues(12.dp),
                verticalArrangement = Arrangement.spacedBy(8.dp)) {
                items(users) { u ->
                    Card(Modifier.fillMaxWidth()) {
                        Column(Modifier.padding(12.dp)) {
                            Row(verticalAlignment = Alignment.CenterVertically) {
                                Text(u.name, fontWeight = FontWeight.Bold, modifier = Modifier.weight(1f))
                                val tone = when (u.role) { "ADMIN" -> IssueRed; "MANAGER" -> Navy; else -> Color.Gray }
                                Surface(color = tone, shape = MaterialTheme.shapes.small) {
                                    Text(u.role.lowercase().replaceFirstChar { it.uppercase() }, color = Color.White,
                                        style = MaterialTheme.typography.labelSmall,
                                        modifier = Modifier.padding(horizontal = 8.dp, vertical = 3.dp))
                                }
                            }
                            Text(u.email, color = Color.Gray, style = MaterialTheme.typography.bodySmall)
                            u.discipline?.let { Text(disciplineLabel(it), color = Color.Gray, style = MaterialTheme.typography.labelSmall) }
                        }
                    }
                }
            }
        }
    }

    if (showAdd) {
        AddUserDialog(backend, onDismiss = { showAdd = false }) {
            showAdd = false; scope.launch { load() }
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
private fun AddUserDialog(backend: Backend, onDismiss: () -> Unit, onCreated: () -> Unit) {
    var name by remember { mutableStateOf("") }
    var email by remember { mutableStateOf("") }
    var password by remember { mutableStateOf("") }
    var role by remember { mutableStateOf("INSPECTOR") }
    var discipline by remember { mutableStateOf("CIVIL") }
    var busy by remember { mutableStateOf(false) }
    var error by remember { mutableStateOf<String?>(null) }
    val scope = rememberCoroutineScope()
    val valid = name.isNotBlank() && email.contains("@") && password.length >= 6

    AlertDialog(
        onDismissRequest = onDismiss,
        title = { Text(tr("Add User")) },
        text = {
            Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                OutlinedTextField(name, { name = it }, label = { Text(tr("Name")) }, singleLine = true)
                OutlinedTextField(email, { email = it }, label = { Text(tr("Email")) }, singleLine = true)
                OutlinedTextField(password, { password = it }, label = { Text(tr("Password (min 6)")) }, singleLine = true)
                EnumDropdown("Role", listOf("INSPECTOR", "MANAGER", "ADMIN"), role,
                    { it.lowercase().replaceFirstChar { c -> c.uppercase() } }) { role = it }
                if (role == "INSPECTOR") {
                    EnumDropdown("Discipline", DISCIPLINES, discipline, ::disciplineLabel) { discipline = it }
                }
                error?.let { Text(it, color = IssueRed, style = MaterialTheme.typography.bodySmall) }
            }
        },
        confirmButton = {
            TextButton(enabled = valid && !busy, onClick = {
                busy = true; error = null
                scope.launch {
                    try {
                        backend.api.createUser(CreateUserBody(name.trim(), email.trim(), password, role,
                            if (role == "INSPECTOR") discipline else null))
                        onCreated()
                    } catch (e: Exception) { error = "Couldn't create the user." }
                    busy = false
                }
            }) { Text(if (busy) "…" else tr("Create")) }
        },
        dismissButton = { TextButton(onClick = onDismiss) { Text(tr("Cancel")) } }
    )
}

// MARK: Small dropdown helpers

@OptIn(ExperimentalMaterial3Api::class)
@Composable
private fun EnumDropdown(label: String, options: List<String>, selected: String, display: (String) -> String, onSelect: (String) -> Unit) {
    var open by remember { mutableStateOf(false) }
    ExposedDropdownMenuBox(expanded = open, onExpandedChange = { open = it }) {
        OutlinedTextField(
            value = display(selected), onValueChange = {}, readOnly = true, label = { Text(label) },
            trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(open) },
            modifier = Modifier.menuAnchor().fillMaxWidth()
        )
        ExposedDropdownMenu(expanded = open, onDismissRequest = { open = false }) {
            options.forEach { o ->
                DropdownMenuItem(text = { Text(display(o)) }, onClick = { onSelect(o); open = false })
            }
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
private fun InspectorDropdown(label: String, options: List<UserRef>, selectedId: String?, onSelect: (String?) -> Unit) {
    var open by remember { mutableStateOf(false) }
    val selectedName = options.firstOrNull { it.id == selectedId }?.name ?: "— none —"
    ExposedDropdownMenuBox(expanded = open, onExpandedChange = { open = it }) {
        OutlinedTextField(
            value = selectedName, onValueChange = {}, readOnly = true, label = { Text(label) },
            trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(open) },
            modifier = Modifier.menuAnchor().fillMaxWidth()
        )
        ExposedDropdownMenu(expanded = open, onDismissRequest = { open = false }) {
            DropdownMenuItem(text = { Text(tr("— none —")) }, onClick = { onSelect(null); open = false })
            options.forEach { u ->
                DropdownMenuItem(text = { Text(u.name) }, onClick = { onSelect(u.id); open = false })
            }
        }
    }
}
